src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1454,6 +1461,7 @@
 
   <Node Name="BoundConstantPattern" Base="BoundPattern">
     <Field Name="Value" Type="BoundExpression"/>
+    <Field Name="ConstantValue" Type="ConstantValue" Null="allow"/>
```

Comment:
Because the value has been converted to the incoming type against which it is matched. For example, if we are matching against `int?`, and the pattern is `3`, the `Value` is the 3 converted to `int?`, i.e. `(int?)3`, which is not a constant. The `ConstantValue` field represents the underlying value of the pattern even if it has been converted.

---
In reply to: [64673953](https://github.com/dotnet/roslyn/pull/11485#discussion_r64673953) [](ancestors = 64673953,64666300)